x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1995,6 +1997,17 @@ private void reloadRemoteClusterCredentials(Settings settingsWithKeystore) {
         future.actionGet();
     }
 
+    @Override
+    public Supplier<Map<String, String>> getLoggingContextSupplier() {
+        return () -> {
+            if (this.securityContext.get() == null || this.securityContext.get().getAuthentication() == null) {
+                return Map.of();
+            }
+            String principal = this.securityContext.get().getAuthentication().getEffectiveSubject().getUser().principal();
+            return Map.of("effective_user", principal);
+        };
+    }
+
```

Comment:
A potential option is to override `SearchOperationListener` and only enable this when we're in a search operation. The risk is that concurrent threads that log will get this context added to them.